### PR TITLE
fix(test-utils): align MockProtocolSim spot_price convention with get_amount_out

### PR DESCRIPTION
**Pull request by pistomat**
## Summary

- **Fix `MockProtocolSim::spot_price()` to match `get_amount_out()` convention**: `spot_price(base, quote)` now returns the marginal rate in the same direction as `get_amount_out(base, quote)`, consistent with other ProtocolSim like UniswapV2State
- **Update all affected test assertions and comments** across `test_utils.rs`, `spot_price.rs`, `pool_depth.rs`, and `most_liquid.rs`
- **Bump `spot_price` from 100 to 200** in `test_binary_search_uses_spot_price_not_effective_price` to avoid a floating-point boundary edge case at exactly 1% impact

## Test plan

- [x] `cargo nextest run -p fynd-core` — all 220 tests pass
- [x] `cargo +nightly fmt --all --check` — clean
- [x] `cargo clippy --workspace --all-targets --all-features -- -D warnings` — clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: fynd-core/src/algorithm/test_utils.rs [hunk 1]
```diff
@@ -88,14 +88,14 @@ impl ProtocolSim for MockProtocolSim {
         self.fee
     }
 
-    /// Note that spot_price does not account for swap direction
+    /// Returns a direction-dependent spot price with fee markup applied.
```
Comment (pistomat) [inline fynd-core/src/algorithm/test_utils.rs:91]: Other ProtocolSims DO have direction-dependent spot price

File: fynd-core/src/algorithm/test_utils.rs [hunk 2]
```diff
@@ -88,14 +88,14 @@ impl ProtocolSim for MockProtocolSim {
...
     fn spot_price(&self, base: &Token, quote: &Token) -> Result<f64, SimulationError> {
         let post_fee_spot_price = self.spot_price.to_f64().unwrap() / (1.0 - self.fee);
         // In order to have asymmetric spot prices based on token order, we define:
         if base.address < quote.address {
-            Ok(1.0 / post_fee_spot_price)
-        } else {
             Ok(post_fee_spot_price)
+        } else {
```
Comment (pistomat) [inline fynd-core/src/algorithm/test_utils.rs:97]: This is fix for the underlying issue

**Review by kayibal — APPROVED** [review]

**Review by dianacarvalho1 — APPROVED** [review]
Thank you @pistomat
